src/Controls/tests/Core.UnitTests/BindableLayoutTests.cs

```diff
@@ -389,6 +389,77 @@ public void ValidateBindableProperties()
 			Assert.Equal(itemTemplateSelector, layout.GetValue(BindableLayout.ItemTemplateSelectorProperty));
 		}
 
+		[Fact]
+		public async Task DoesNotLeakChildListeningToBindingContextOnRemove()
+		{
+			var myViewModel = new MyViewModel();
+			var list = new ObservableCollection<MyViewModel> { myViewModel };
+			
+			var triggeredCount = 0;
+			var itemTemplate = new DataTemplate(() => new MyViewModelBoundComponent(onTextChangedCallback: () => triggeredCount++));
+
+			var layout = new StackLayout { IsPlatformEnabled = true };
+			BindableLayout.SetItemTemplate(layout, itemTemplate);
+			BindableLayout.SetItemsSource(layout, list);
+			
+			list.RemoveAt(0);
+			
+			// Run GC
+			await Task.Yield();
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+			
+			// The component should be gone
+			Assert.Equal(0, MyViewModelBoundComponent.InstanceCount);
```

Comment:
Can we instead create a `List<WeakReference>` that contains each `MyViewModelBoundComponent` object, and assert `IsAlive` is `false`? Then you wouldn't need a manual `InstanceCount`.

Example:
https://github.com/dotnet/maui/blob/aad88d6f01a9e244df189a2031d6b1df7be4a019/src/Controls/tests/DeviceTests/Memory/MemoryTests.cs#L79